src/Reflectivity-VariableBreakpoint-Extensions/ClyVariableBreakpointCommand.class.st

```diff
@@ -0,0 +1,48 @@
+"
+Commands to install a variable breakpoint on the selected variable in the source code.
+If the variable is a temporary, installs the variable breakpoint on all variable nodes refering to that variable name in the method.
+If the variable is an intance variable, installs the breakpoint on all methods using the variable.
+"
+Class {
+	#name : #ClyVariableBreakpointCommand,
+	#superclass : #ClyDebuggingCommand,
+	#category : #'Reflectivity-VariableBreakpoint-Extensions'
+}
+
+{ #category : #testing }
+ClyVariableBreakpointCommand class >> canBeExecutedInContext: aSourceCodeContext [
+	^ self isAbstract not
+		and: [ aSourceCodeContext class = ClyMethodSourceCodeContext
```

Comment:
What you are trying to disable here?
Why it is not suitable for any source code context?

Reply:
I will check again, but to me it was buggy: the source code context are not always consistent.

Reply:
Yes, so because sometimes it is a `ClyMethodContextOfFullBrowser` (when clicking on a variable in the var view). I really do not understand it, how to use commands in Calypso is cryptic. If I do not filter like this, it tries to call the source node on that context and it fails.

Reply:
It looks like a bug. I will look when you merge

Reply:
Not a bug. You should use #lastSelectedSourceNode instead of #selectedSourceNode. But I understand the confusion. 
#selectedSourceNode is internal method of ClySourceNodeContext. But the general API to access objects from context always follow multiple selection idea. When command needs just one object it should be explicit first or last selected object. 
For example in method context you also can get lastSelectedSourceNode which is an ast of last selected method. And it is a reason why you see the method context here: it is compatible to source code commands.

Reply:
I think when you will fix it you will get another cool place for new commands - try it from class definition on variable names.

Reply:
It works :) So I removed the class check and use lastSelectedNode as you suggested.